dsc_lib/src/configure/mod.rs

```diff
@@ -161,6 +162,36 @@ fn add_metadata(kind: &Kind, mut properties: Option<Map<String, Value>> ) -> Res
     Ok(serde_json::to_string(&properties)?)
 }
 
+fn check_security_context(config: &Configuration) -> Result<(), DscError> {
```

Comment:
is the entire `Configuration` needed by `check_security_context` or can just the `metadata` be passed in here?

Reply:
good question.  `Configuration` is passed by reference, so there's no penalty.  It could just require the `metadata` which might make it a bit more clear where the check is happening.  I'll make the change.